lib/src/Uncertainty/Distribution/PointConditionalDistribution.cxx

```diff
@@ -0,0 +1,763 @@
+    const Point mY(conditioningValues_ - mu.select(conditioningIndices_));
+    const Scalar dy = mY.dot(distribution_.getMarginal(conditioningIndices_).getCovariance().solveLinearSystem(mY));
+    const Scalar nu = p_student->getNu();
+    const UnsignedInteger py = conditioningIndices_.getSize();
+    C = CovarianceMatrix((C * std::sqrt((nu + dy) / (nu + py))).getImplementation());
+    Point sigma(getDimension());
+    CorrelationMatrix R(getDimension());
+    for (UnsignedInteger i = 0; i < getDimension(); ++ i)
+    {
+      const Scalar cii = C(i, i);
+      sigma[i] = std::sqrt(cii);
+      for (UnsignedInteger j = 0; j < i; ++ j)
+        R(i, j) = C(i, j) / (sigma[i] * sigma[j]);
+    }
+    simplified = Student(nu + py, mu, sigma, R);
+    return true;
+  }
+
+  return false;
+}
+
+Distribution PointConditionalDistribution::getSimplifiedVersion() const
+{
+  return useSimplifiedVersion_ ? simplifiedVersion_ : *this;
+}
+
+/* Compute the numerical range of the distribution given the parameters values */
+void PointConditionalDistribution::computeRange()
+{
+  if (useSimplifiedVersion_)
+    setRange(simplifiedVersion_.getRange());
+  else
+    setRange(distribution_.getRange().getMarginal(nonConditioningIndices_));
+}
+
+/* Get the support of a distribution that intersects a given interval */
+Sample PointConditionalDistribution::getSupport(const Interval & interval) const
+{
+  if (interval.getDimension() != getDimension()) throw InvalidArgumentException(HERE) << "Error: the given interval has a dimension that does not match the distribution dimension.";
+  if (!isDiscrete()) throw NotDefinedException(HERE) << "Error: the support is defined only for discrete distributions.";
+  Collection<Sample> marginalSupport(getDimension());
+  Indices marginalSize(getDimension());
+  for (UnsignedInteger j = 0; j < getDimension(); ++ j)
+  {
+    marginalSupport[j] = distribution_.getMarginal(nonConditioningIndices_[j]).getSupport(interval.getMarginal(j));
+    marginalSize[j] = marginalSupport[j].getSize();
+  }
+  IndicesCollection tuples(Tuples(marginalSize).generate());
+  Sample support(0, getDimension());
+  for (UnsignedInteger i = 0; i < tuples.getSize(); ++ i)
+  {
+    Point x(getDimension());
+    for (UnsignedInteger j = 0; j < getDimension(); ++ j)
+      x[j] = marginalSupport[j](tuples(i, j), 0);
+    if (computePDF(x) > 0.0)
+      support.add(x);
+  }
+  return support;
+}
+
+/* Get one realization of the distribution */
+Point PointConditionalDistribution::getRealization() const
+{
+  if (useSimplifiedVersion_)
+    return simplifiedVersion_.getRealization();
+
+  const UnsignedInteger dimension = getDimension();
+  if (dimension <= ResourceMap::GetAsUnsignedInteger("PointConditionalDistribution-SmallDimension"))
+  {
+    // the ratio of uniforms method, see https://en.wikipedia.org/wiki/Ratio_of_uniforms
+    // r is a free parameter, could be optimized to maximize the acceptance ratio
+    const Scalar r = 1.0;
```

Comment:
If we decide to optimize wrt r we have to put all these things in the evaluation of a new function, and many things can be reused from one computation to the other. In particular, the bound on U is located at the same place regardless of the value of r, even if the value of this bound depends on r.